modules/apps/foundation/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/util/comparator/OrganizationMemberNameComparator.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.users.admin.web.util.comparator;
+
+import com.liferay.portal.kernel.model.Organization;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.util.OrderByComparator;
+
+/**
+ * @author Pei-Jung Lan
+ */
+public class OrganizationMemberNameComparator
+	extends OrderByComparator<Object> {
+
+	public static final String ORDER_BY_ASC = "name ASC";
+
+	public static final String ORDER_BY_DESC = "name DESC";
+
+	public static final String[] ORDER_BY_FIELDS = {"name"};
+
+	public OrganizationMemberNameComparator() {
+		this(false);
+	}
+
+	public OrganizationMemberNameComparator(boolean ascending) {
+		_ascending = ascending;
+	}
+
+	@Override
+	public int compare(Object object1, Object object2) {
+		String name1;
+		String name2;
+
+		if (object1 instanceof Organization) {
```

Comment:
Seeing this method made me think that instead of doing this in different places of the code it would be better to create an interface that is common for User and Organization. An example of this is RepositoryEntry, which is used in Document Library. By having this interface they don't have to return List<Object> from their getFoldersAndDocuments() methods, but rather List<RepositoryEntry>.

Similarly, for this you could have an OrganizationEntry interface (or some other similar name) and then return List<OrganizationEntry>. That interface could have all the shared getters and also a getName() method which in User could be implemented as returning getLastName(). This class would then be OrganizationEntryNameComparator and would not have a need to have this if clause. Also it would not need any modification if in the future a different type of entry is added.

What do you think?